### ISSUE TITLE
merge two networks

### ISSUE THREAD

**Issue by guofuzheng** [keras-team/keras#11359]
I want to fuse two different networks with different weights. And the weights computed by another network. I could not find  any function or api to fuse them. Should I multiply and add them？Could you help me? Thank you for your help.

**Comment by farizrahman4u** [comment]
What do you mean by fuse? Given 2 models of the save config, you can create a "mean" model like this:

```ptyhon
model1 = ...
model2 = ...

W1 = model1.get_weights()
W2 = model2.get_weights()

W_mean = []
for w1, w2 in zip(W1, W2):
    avg = np.mean([w1, w2])
    W_mean.append(avg)

model3 = keras.models.load_model(model1.get_config())  # create a copy
model3.set_weights(W_mean)
```

**Comment by guofuzheng** [comment]
![image](https://user-images.githubusercontent.com/25239334/46913165-ca516a00-cfba-11e8-8d3c-4a2cb29b2e04.png)
@farizrahman4u Hi, I want to use the network 'weight compute' to compute the weights to fuse the network1 and network2. For example the outputs of the network 'weight compute' are (w1 and w2). I wanna w1 multiply the output of network1 and w2 multiply the output2 of network2 and add the results.Then connect to layer Dense and use softmax to classify .I want to merge two networks by different weights( not fixed weights) when compute different samples. What should I do ？Thank you for your help.

**Comment by farizrahman4u** [comment]
Thanks for the detailed description and diagram.. Here you go:

```python
net1 = ....
net2 = ....
weight_compute = ...

inp = Input(....)   # (batch_size, input_dim)

net1_out = net1(inp)   # (batch_size, dim)
net2_out = net2(inp)   # (batch_size, dim)
w1_w2 = weight_compute(inp)  # (batch_size, 2)
w1 = Lambda(lambda x: x[:, :1], output_shape=lambda s: (s[0], 1))(w1_w2)  # (batch_size, 1)
w2 = Lambda(lambda x: x[:, 1:], output_shape=lambda s: (s[0], 1))(w1_w2)  # (batch_size, 1)
net1_w1 = multiply([net1_out, w1])
net2_w2 = multiply([net2_out, w2])
weighted_sum = add([net1_w1, net2_w2])
dense_out = Dense(....)(weighted_sum)
softmax_out = Activation('softmax')(dense_out)

model = Model(inp, softmax_out)
```
